e2e/libs/utils/common_utils.py

```diff
@@ -0,0 +1,139 @@
+import os
+import requests
+import warnings
+import string
+import random
+import time
+import socket
+import longhorn
+
+from utils import config_utils
+from kubernetes import config, client
+from kubernetes.client import Configuration
+from longhorn import from_env
+
+PORT = ":9500"
+MAX_SUPPORT_BUNDLE_NUMBER = 20
+RETRY_EXEC_COUNTS = 150
+RETRY_INTERVAL = 1
+RETRY_INTERVAL_LONG = 2
+
+def k8s_core_api():
+    c = Configuration()
+    c.assert_hostname = False
+    Configuration.set_default(c)
+    config.load_incluster_config()
+    core_api = client.CoreV1Api()
+    return core_api
+
+def k8s_cr_api():
+    c = Configuration()
+    c.assert_hostname = False
+    Configuration.set_default(c)
+    config.load_incluster_config()
+    cr_api = client.CustomObjectsApi()
+    return cr_api
+
+def get_longhorn_api_client():
+    for i in range(RETRY_EXEC_COUNTS):
+        try:
+            config.load_incluster_config()
+            ips = get_mgr_ips()
+
+            # check if longhorn manager port is open before calling get_client
+            for ip in ips:
+                sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+                mgr_port_open = sock.connect_ex((ip, 9500))
+
+                if mgr_port_open == 0:
+                    client = 'http://' + ip + PORT + '/' #get_client(ip + PORT)
+                    break
+            return client
+        except Exception:
+            time.sleep(RETRY_INTERVAL)
+
+def get_mgr_ips():
+    ret = k8s_core_api().list_pod_for_all_namespaces(
+            label_selector="app=longhorn-manager",
+            watch=False)
+    mgr_ips = []
+    for i in ret.items:
+        mgr_ips.append(i.status.pod_ip)
+    return mgr_ips
+
+def get_client(address):
+    url = 'http://' + address + '/v1/schemas'
+    c = longhorn.from_env(url=url)
+    return c
+
+def generate_volume_name():
+    return "vol-" + \
+        ''.join(random.choice(string.ascii_lowercase + string.digits)
+                for _ in range(6))
+
+def get_longhorn_client():
+    # manually expose longhorn client node port
+    # otherwise the test is needed to be run in in-cluster environment
+    # to access longhorn manager cluster ip
+    longhorn_client_url = get_longhorn_api_client() #config_utils.Config.get("LONGHORN_CLIENT_URL")
+    print(f'longhorn_client_url=>{longhorn_client_url}')
```

Comment:
Remove if this is not an intended log.

Reply:
I've confirmed everything except `longhorn.py`, modified it to use logging or removed.
![image](https://github.com/longhorn/longhorn-tests/assets/17548901/9213de7a-0ac9-42c2-8649-2e8d59de3515)